server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -3263,73 +3273,89 @@ SnapshotsInProgress computeUpdatedState() {
             return existing;
         }
 
-        void completeWithUpdatedState(SnapshotsInProgress snapshotsInProgress) {
-            if (updatesByRepo.isEmpty() == false) {
-                final var result = new ShardSnapshotUpdateResult(initialState.metadata(), snapshotsInProgress);
-                try (
-                    var onCompletionRefs = new RefCountingRunnable(
-                        () -> completionHandler.handleCompletion(result, newlyCompletedEntries, updatesByRepo.keySet())
-                    )
-                ) {
-                    for (final var taskContext : batchExecutionContext.taskContexts()) {
-                        if (taskContext.getTask() instanceof ShardSnapshotUpdate task) {
-                            final var ref = onCompletionRefs.acquire();
-                            taskContext.success(() -> {
-                                try (ref) {
-                                    task.listener.onResponse(result);
-                                }
-                            });
-                        }
+        /**
+         * Reply to all the listeners of the {@link ShardSnapshotUpdate} tasks in {@link #batchExecutionContext}; then run the final
```

Comment:
Technically we're not completing the listeners here, we're just queueing up those completions to happen if the cluster state update succeeds.

Reply:
Ah. I got fooled by the success() method name 😌 Thanks, updated 👍